waku/v2/protocol/waku_rln_relay/waku_rln_relay_utils.nim

```diff
@@ -952,67 +962,85 @@ proc appendRLNProof*(rlnPeer: WakuRLNRelay, msg: var WakuMessage,
 
-proc subscribeToMemberRegistrations(web3: Web3, 
-                                    contractAddress: Address,
-                                    fromBlock: string = "0x0",
-                                    handler: GroupUpdateHandler): Future[Subscription] {.async, gcsafe.} =
-  ## subscribes to member registrations, on a given membership group contract
-  ## `fromBlock` indicates the block number from which the subscription starts
-  ## `handler` is a callback that is called when a new member is registered
-  ## the callback is called with the pubkey and the index of the new member
-  ## TODO: need a similar proc for member deletions
-  var contractObj = web3.contractSender(MembershipContract, contractAddress)
-
-  let onMemberRegistered = proc (pubkey: Uint256, index: Uint256) {.gcsafe.} =
-    debug "onRegister", pubkey = pubkey, index = index
-    var groupUpdateRes: RlnRelayResult[void]
-    try:
-      groupUpdateRes = handler(pubkey, index)
-    except Exception as err:
-      error "failed to handle group update", err = err.msg
-    if groupUpdateRes.isErr():
-      error "Error handling new member registration", err=groupUpdateRes.error()
-
-  let onError = proc (err: CatchableError) =
-    error "Error in subscription", err=err.msg
-
-  return await contractObj.subscribe(MemberRegistered,
-                                     %*{"fromBlock": fromBlock, "address": contractAddress},
-                                     onMemberRegistered,
-                                     onError)
+proc parse*(event: type MemberRegistered, 
+            log: JsonNode): RlnRelayResult[MembershipTuple] =
+  ## parses the `data` parameter of the `MemberRegistered` event
+  ## returns an error if it cannot parse the `data` parameter
+  var pubkey: UInt256
+  var index: UInt256
+  var data: string
+  try:
+    data = strip0xPrefix(log["data"].getStr())
+  except CatchableError as err:
+    return err("failed to parse the data field of the MemberRegistered event: " & err.msg)
+  var offset = 0
+  try:
+    offset += decode(data, offset, pubkey)
+    offset += decode(data, offset, index)
+    return ok((index: index.toMembershipIndex(), 
+               idComm: pubkey.toIDCommitment()))
+  except:
+    return err("failed to parse the data field of the MemberRegistered event")
+
+type BlockTable = OrderedTable[BlockNumber, seq[MembershipTuple]]
+proc getHistoricalEvents*(ethClientUri: string,
+                          contractAddress: Address,
+                          fromBlock: string = "0x0",
+                          toBlock: string = "latest"): RlnRelayResult[BlockTable] {.async, gcsafe.} =
+  ## returns a table that maps block numbers to the list of members registered in that block
+  ## returns an error if it cannot retrieve the historical events
+  let web3 = await newWeb3(ethClientUri)
+  let contract = web3.contractSender(MembershipContract, contractAddress)
+  let historicalEvents = await contract.getJsonLogs(MemberRegistered,
```

Comment:
Please add a comment about this line.

Reply:
Addressed in [8b0ff46](https://github.com/status-im/nwaku/pull/1349/commits/8b0ff463b4b8f307d33475e365ca1d0c389e39d7)